gitlab/resource_gitlab_project_test.go

```diff
@@ -509,6 +509,64 @@ func TestAccGitlabProject_importURL(t *testing.T) {
 	})
 }
 
+func TestAccGitlabProject_importURLMirrored(t *testing.T) {
+	// Since we do some manual setup in this test, we need to handle the test skip first.
+	if os.Getenv(resource.TestEnvVar) == "" {
+		t.Skip(fmt.Sprintf("Acceptance tests skipped unless env '%s' set", resource.TestEnvVar))
+	}
+
+	client := testAccProvider.Meta().(*gitlab.Client)
+	rInt := acctest.RandInt()
+
+	// Create a base project for importing.
+	baseProject, _, err := client.Projects.CreateProject(&gitlab.CreateProjectOptions{
+		Name:       gitlab.String(fmt.Sprintf("base-%d", rInt)),
+		Visibility: gitlab.Visibility(gitlab.PublicVisibility),
+	})
+	if err != nil {
+		t.Fatalf("failed to create base project: %v", err)
+	}
+
+	defer client.Projects.DeleteProject(baseProject.ID)
+
+	// Add a file to the base project, for later verifying the import.
+	_, _, err = client.RepositoryFiles.CreateFile(baseProject.ID, "foo.txt", &gitlab.CreateFileOptions{
+		Branch:        gitlab.String("master"),
+		CommitMessage: gitlab.String("add file"),
+		Content:       gitlab.String(""),
+	})
+	if err != nil {
+		t.Fatalf("failed to commit file to base project: %v", err)
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGitlabProjectConfigImportURLMirrored(rInt, baseProject.HTTPURLToRepo),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("gitlab_project.imported", "import_url", baseProject.HTTPURLToRepo),
+					resource.TestCheckResourceAttr("gitlab_project.mirrored", "mirrored", "true"),
+					resource.TestCheckResourceAttr("gitlab_project.mirror_target_builds", "mirror_target_builds", "true"),
+
```

Comment:
The new test looks fishy; it has some assertions on resources that are not being created by the test config. Unfortunately we can't run the acceptance tests in the GitHub workflow until #495 and #494 are resolved.

```suggestion
```

Reply:
@armsnyder I'm not sure I follow...

I thought these test the values of the properties adding in this PR.

Is there a different way to check these are set on the resource properly or am I missing something?

The properties added are here:
https://github.com/cnuss/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_project_test.go#L904-L906

Reply:
No worries. I was referring to the resource name, which is `gitlab_project.imported` (see: https://github.com/cnuss/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_project_test.go#L901). However in your `resource.TestCheckResourceAttr` call you're using a resource name of `gitlab_project.mirrored`. My suggestion to remove these three lines is because the Terraform test framework will fail automatically if the resulting attributes do not match what was configured.

Reply:
Ahh... then this would also be correct if I wanted to do an explicit check? Sounds like you have a preference for the implicit check?

```
					resource.TestCheckResourceAttr("gitlab_project.imported", "import_url", baseProject.HTTPURLToRepo),
					resource.TestCheckResourceAttr("gitlab_project.imported", "mirrored", "true"),
					resource.TestCheckResourceAttr("gitlab_project.imported", "mirror_target_builds", "true"),
```

Reply:
Yup, either way is fine. I was just giving you the lazier option. 😄

Reply:
Pushed the non-lazy option ;) Awaiting #494 !

Reply:
@cnuss #494 was closed. Is this ready for merge?

Reply:
Can we resolve this conversation?

Reply:
Yup fixed, resolving convo